comet/archiver.py

```diff
@@ -1,41 +1,61 @@
 """
 Archiver for CoMeT.
 
-Moves comet broker dumps to a database
+Move comet broker data from redis to a mysql database
 """
 
-import asyncio
 import datetime
 import logging
 import orjson as json
-from pathlib import Path
-import signal
+import random
+import redis
+import time
 
 import chimedb.core as chimedb
 import chimedb.dataset as db
 
 from . import Manager, CometError, __version__
 from .manager import TIMESTAMP_FORMAT
 
-
 logger = logging.getLogger(__name__)
-logger.setLevel("INFO")
 
 
 class Archiver:
-    """Main class to run the comet archiver."""
-
-    def __init__(self, data_dump_path, scrape_interval, broker_host, broker_port):
+    """
+    Main class to run the comet archiver.
+
+    New data that should be archived is expected in a list in redis. The archiver currently
+    monitors the lists `registered_dataset` and `registered_state`. If a state or dataset
+    referenced in any new item, the archiver pushes the item back on the list for later.
+    """
+
+    def __init__(
+        self,
+        broker_host,
+        broker_port,
+        redis_host,
+        redis_port,
+        log_level,
+        failure_wait_time,
+    ):
+        logger.setLevel(log_level)
+
+        # convert ms to s
+        self.failure_wait_time = failure_wait_time / 1000
 
         startup_time = datetime.datetime.utcnow()
-        config = {"data_dump_path": data_dump_path, "scrape_interval": scrape_interval}
-        self.dir = data_dump_path
-        self.interval = scrape_interval
+        config = {
+            "broker_host": broker_host,
+            "broker_port": broker_port,
+            "redis_host": redis_host,
+            "redis_port": redis_port,
+            "log_level": log_level,
+            "failure_wait_time": failure_wait_time,
+        }
 
         manager = Manager(broker_host, broker_port)
         try:
-            manager.register_start(startup_time, __version__)
-            manager.register_config(config)
+            manager.register_start(startup_time, __version__, config)
```

Comment:
Now register_start also registers the config?

Reply:
Yeah, we decided it makes no sense you have to call two separate functions. I am trying to make it support the old API for a while though.